src/lib/src/common/geometry/interpolate.cpp

```diff
@@ -70,45 +65,39 @@ interpolate_z_triangles(const double x,
  *
  * @param x X coordinate of the point
  * @param y Y coordinate of the point
- * @param p1 A vector of doubles, length 3
- * @param p2 A vector of doubles, length 3
- * @param p3 A vector of doubles, length 3
- * @param p4 A vector of doubles, length 3
+ * @param p1 Point 1 instance
+ * @param p2 Point 2 instance
+ * @param p3 Point 3 instance
+ * @param p4 Point 4 instance
  * @return double
  */
 double
 interpolate_z_4p_regular(const double x,
                          const double y,
-                         const std::array<double, 3> &p1,
-                         const std::array<double, 3> &p2,
-                         const std::array<double, 3> &p3,
-                         const std::array<double, 3> &p4)
+                         const Point &p1,
+                         const Point &p2,
+                         const Point &p3,
+                         const Point &p4)
```

Comment:
It feels like there is some loss of generality with this change. Putting a hard requirement on taking `Point`s will always mean creating some basically duplicated objects (duplicating the data from a tuple or array into a Point). Maybe that doesn't matter, maybe it's a micro-performance hit that becomes relevant at some point (no pun intended). It is certainly nicer to read the code with the `Point` though :)